### ISSUE TITLE
Issue during in-painting classification

### ISSUE THREAD

**Issue by mcianfrocco** [rocketmlhq/rmldnn#3]
Hello - I'm working through the tutorial for in-painting and I've successfully completed the feature extraction training. 

![Screen Shot 2022-04-25 at 4 11 13 PM](https://user-images.githubusercontent.com/1464469/165167094-1575ac0e-dcfb-4e65-b380-29f5d474a64a.png)

However, even though the classification training finishes successfully, I see that the loss balloons to a large number. Do you have any sense of what this could mean? 
 
Here is the output from training the classifier: 
```
$ cat rmldnnClass.out | grep Accuracy
[2022-Apr-25 15:28:10.260352] Eval  Epoch [  1/100] : Batch [    1/    1] (Loss: 11.9174) | Accuracy: 0.618
[2022-Apr-25 15:28:26.116163] Eval  Epoch [  2/100] : Batch [    1/    1] (Loss: 71.4548) | Accuracy: 0.634
[2022-Apr-25 15:28:43.123124] Eval  Epoch [  3/100] : Batch [    1/    1] (Loss: 436.4486) | Accuracy: 0.605
[2022-Apr-25 15:28:58.762846] Eval  Epoch [  4/100] : Batch [    1/    1] (Loss: 4738.9883) | Accuracy: 0.601
[2022-Apr-25 15:29:14.939691] Eval  Epoch [  5/100] : Batch [    1/    1] (Loss: 22715.8613) | Accuracy: 0.622
[2022-Apr-25 15:29:30.472585] Eval  Epoch [  6/100] : Batch [    1/    1] (Loss: 117077.8047) | Accuracy: 0.626
[2022-Apr-25 15:29:45.980462] Eval  Epoch [  7/100] : Batch [    1/    1] (Loss: 1560777.3750) | Accuracy: 0.599
[2022-Apr-25 15:30:01.238265] Eval  Epoch [  8/100] : Batch [    1/    1] (Loss: 10272561.0000) | Accuracy: 0.628
[2022-Apr-25 15:30:16.822373] Eval  Epoch [  9/100] : Batch [    1/    1] (Loss: 106860592.0000) | Accuracy: 0.537
[2022-Apr-25 15:30:32.327156] Eval  Epoch [ 10/100] : Batch [    1/    1] (Loss: 492095744.0000) | Accuracy: 0.598
[2022-Apr-25 15:30:38.241459] Eval  Epoch [ 11/100] : Batch [    1/    1] (Loss: 1098736640.0000) | Accuracy: 0.593
[2022-Apr-25 15:30:42.281468] Eval  Epoch [ 12/100] : Batch [    1/    1] (Loss: 1204668032.0000) | Accuracy: 0.619
[2022-Apr-25 15:30:46.724595] Eval  Epoch [ 13/100] : Batch [    1/    1] (Loss: 1348660480.0000) | Accuracy: 0.639
[2022-Apr-25 15:30:51.073770] Eval  Epoch [ 14/100] : Batch [    1/    1] (Loss: 1276815488.0000) | Accuracy: 0.639
[2022-Apr-25 15:30:55.363943] Eval  Epoch [ 15/100] : Batch [    1/    1] (Loss: 1593674496.0000) | Accuracy: 0.639
```

Information on training: 
```
[2022-Apr-25 15:27:20.244494] RocketML : dnn
[2022-Apr-25 15:27:20.244561] rocketml 1.0.0 (Linux-5.3.0-1031-azure ) (Apr 14 2022 22:57:24) (git:master rev:b822b0c)
[2022-Apr-25 15:27:20.244570] RocketML : 4 MPI processes
[2022-Apr-25 15:27:20.244591]                     ___        __
[2022-Apr-25 15:27:20.244599]                    /\_ \      /\ \
[2022-Apr-25 15:27:20.244605]  _ __    ___ ___   \//\ \     \_\ \     ___      ___
[2022-Apr-25 15:27:20.244611] /\`'__\ /' __` __`\  \ \ \    /'__ \  /' _ `\  /' _ `\
[2022-Apr-25 15:27:20.244617] \ \ \/  /\ \/\ \/\ \  \_\ \_ /\ \_\ \ /\ \/\ \ /\ \/\ \
[2022-Apr-25 15:27:20.244623]  \ \_\  \ \_\ \_\ \_\ /\____\\ \___,_\\ \_\ \_\\ \_\ \_\
[2022-Apr-25 15:27:20.244628]   \/_/   \/_/\/_/\/_/ \/____/ \/__,_ / \/_/\/_/ \/_/\/_/
[2022-Apr-25 15:27:20.244634]
[2022-Apr-25 15:27:20.244639]             (C) 2022 RocketML, Inc. All rights reserved.
[2022-Apr-25 15:27:20.244645]
[2022-Apr-25 15:27:20.244650] |-------------------------------------------------------------------|
[2022-Apr-25 15:27:20.244656] |                   RocketML Deep Neural Networks                   |
[2022-Apr-25 15:27:20.244661] |-------------------------------------------------------------------|
[2022-Apr-25 15:27:20.244667] | More info: https://github.com/rocketmlhq/rmldnn                   |
[2022-Apr-25 15:27:20.244673] | License  : https://github.com/rocketmlhq/rmldnn/blob/main/LICENSE |
[2022-Apr-25 15:27:20.244678] | Contact  : support@rocketml.net                                   |
[2022-Apr-25 15:27:20.244684] |-------------------------------------------------------------------|
[2022-Apr-25 15:27:20.244689]
[2022-Apr-25 15:27:20.410407] ----------------- Device(s) -----------------
[2022-Apr-25 15:27:20.410433]    CUDA:0
[2022-Apr-25 15:27:20.410441]    CUDA:1
[2022-Apr-25 15:27:20.410447]    CUDA:2
[2022-Apr-25 15:27:20.410452]    CUDA:3
[2022-Apr-25 15:27:20.410458] ---------------------------------------------
[2022-Apr-25 15:27:20.583516] -------------- Gradient reducer -------------
[2022-Apr-25 15:27:20.583552]    Type: oneshot
[2022-Apr-25 15:27:20.584248] ---------------------------------------------
[2022-Apr-25 15:27:20.584259] --------------- Neural Network --------------
[2022-Apr-25 15:27:20.584264]    Model name          : model_2
[2022-Apr-25 15:27:20.584268]    Total parameters    : 23796744
[2022-Apr-25 15:27:20.584388]    Trainable parameters: 262152
[2022-Apr-25 15:27:20.584492]    Num of operations   : 176
[2022-Apr-25 15:27:20.584494] ---------------------------------------------
[2022-Apr-25 15:27:20.586453] Loading model checkpoint from file: ./model_checkpoints/model_checkpoint_100.pt
[2022-Apr-25 15:27:22.166056]    Skipping layer dense_1: not found in model
[2022-Apr-25 15:27:22.166088]    Skipping parameter dense_1.weight: not found in model
[2022-Apr-25 15:27:22.166096]    Skipping parameter dense_1.bias: not found in model
[2022-Apr-25 15:27:29.930647] ------------- TAO configuration -------------
[2022-Apr-25 15:27:29.930667]    Optimization algo:  bqnls
[2022-Apr-25 15:27:29.930671]    Max iterations:     10
[2022-Apr-25 15:27:29.930676]    Max func evals:     4000
[2022-Apr-25 15:27:29.930678]    Absolute tolerance: 1e-08
[2022-Apr-25 15:27:29.930690]    Relative tolerance: 1e-08
[2022-Apr-25 15:27:29.930693]    Line search algo:   more-thuente
[2022-Apr-25 15:27:29.930696] ---------------------------------------------
[2022-Apr-25 15:27:29.930735] -------------------- Loss -------------------
[2022-Apr-25 15:27:29.930744]    Function    : NLL (Negative Log-Likelihood)
[2022-Apr-25 15:27:29.930750]    Weight      : None
[2022-Apr-25 15:27:29.930759]    Ignore index: None
[2022-Apr-25 15:27:29.930764] ---------------------------------------------
[2022-Apr-25 15:27:29.930802] Discovering training input images...
[2022-Apr-25 15:27:34.766006] Pre-loading training input images...
[2022-Apr-25 15:27:44.770813]    62% (ETA 6.1s)
[2022-Apr-25 15:27:50.942811] Discovering training labels...
[2022-Apr-25 15:27:50.956921] Number of class labels: 8
[2022-Apr-25 15:27:50.971304] Discovering test input images...
[2022-Apr-25 15:27:50.986674] Pre-loading test input images...
[2022-Apr-25 15:27:52.942285] Discovering test labels...
[2022-Apr-25 15:27:52.956547] Number of class labels: 8
```

**Comment by mcianfrocco** [comment]
Thank you - yes, this led to a more expected trend for the loss. Here is the training loss and the output accuracy on the test subset. 

This appears to be comparable to the tutorial, do you agree? 
<img width="631" alt="Screen Shot 2022-04-26 at 2 15 31 PM" src="https://user-images.githubusercontent.com/1464469/165366719-d579017d-34c1-496b-8e82-ed55584773a3.png">

<img width="635" alt="Screen Shot 2022-04-26 at 2 21 02 PM" src="https://user-images.githubusercontent.com/1464469/165366718-0b173271-b1e0-4c1c-a6f0-70eb7b4b6e19.png">

**Comment by ssbotelh** [comment]
Hi Michael,
The loss seems decent, but the test accuracy is quite below what I expected. Did you also use the new network that I committed in this last update? I unfroze several layers in the network, which showed to improve the classifier accuracy.

**Comment by mcianfrocco** [comment]
Hi Sergio - I apologize, yes I forgot to download the new network json file. This gave improved results, thank you! 


<img width="633" alt="Screen Shot 2022-04-26 at 10 14 57 PM" src="https://user-images.githubusercontent.com/1464469/165425843-e7fe06e7-1f3a-4b83-a625-2fcd1f5a45ae.png">
<img width="714" alt="Screen Shot 2022-04-26 at 10 14 04 PM" src="https://user-images.githubusercontent.com/1464469/165425845-97fcab41-1fc5-4a75-8dec-23157350a3d9.png">

**Comment by ssbotelh** [comment]
Excellent! I'm glad this worked end-to-end.
